### PR TITLE
feat(umbrella): Deprecate google-api-client in favor of individual service-specific gems

### PR DESCRIPTION
**Pull request by dazuma**
This PR prepares `google-api-client` to be rereleased as "deprecated". We can release it after the initial release of all the individual client libraries (currently scheduled for Jan 7).

Specifically:

* Rewrote `OVERVIEW.md` to include just a deprecation notice and a FAQ.
* Added a post-install message noting the deprecation.
* Removed all the documentation from the gem. (The documentation has already been replicated/moved elsewhere.)
* Removed some rogue/orphaned `synth.metadata` files that were generated in the wrong place by a synthtool bug.

### CODE REVIEW

File: google-api-client/google-api-client.gemspec [hunk 1]
```diff
@@ -13,9 +13,16 @@ Gem::Specification.new do |gem|
   gem.metadata      = {
     "documentation_uri" => "https://googleapis.dev/ruby/google-api-client/v#{Google::Apis::VERSION}"
   }
+
+  gem.post_install_message = <<~MESSAGE
+    ** The google-api-client gem is deprecated and will not be updated further. **
```
Comment (quartzmo) [inline google-api-client/google-api-client.gemspec:18]: This statement is different from the OVERVIEW, which states:

> This library is **deprecated** and will likely not be updated in the future.

It seems to me that the message should be identical.

File: google-api-client/OVERVIEW.md [hunk 2]
```diff
@@ -1,36 +1,45 @@
-# Google API Client Library for Ruby Docs
+# Monolithic Client for Google APIs (DEPRECATED)
 
-The Google API Client Library for Ruby is designed for Ruby
-client-application developers. It offers simple, flexible
-access to many Google APIs.
+This library includes REST clients for all Discovery-based Google APIs as of December 2020.
```
Comment (quartzmo) [inline google-api-client/OVERVIEW.md:3]: Link "Discovery" to `https://developers.google.com/discovery` or similar?

File: google-api-client/OVERVIEW.md [hunk 3]
```diff
@@ -1,36 +1,45 @@
...
-- Call Google APIs simply
-- Handle Auth with fewer lines of code
-- Use standard tooling for installation
+## FAQ
 
-## Documentation
+### Why is this gem deprecated?
 
-Learn how to use the Google API Client Library for Ruby with these guides:
+The `google-api-client` gem is a 75 megabyte installation including client classes for more than 300 services. Although the library is designed so that only the classes being used are loaded into memory, it still takes a nontrivial amount of space on disk (and in deployments), and can take a while to install. Additionally, the documentation is so large that rubydoc.info often fails to build it.
```
Comment (quartzmo) [inline google-api-client/OVERVIEW.md:11]: If the documentation is removed, should this be past tense? ("was so large... failed..")

File: google-api-client/OVERVIEW.md [hunk 4]
```diff
@@ -1,36 +1,45 @@
-# Google API Client Library for Ruby Docs
+# Monolithic Client for Google APIs (DEPRECATED)
 
-The Google API Client Library for Ruby is designed for Ruby
-client-application developers. It offers simple, flexible
-access to many Google APIs.
+This library includes REST clients for all Discovery-based Google APIs as of December 2020.
 
-## Features
+This library is **deprecated** and will likely not be updated in the future. Instead, please install the gem corresponding to the specific service you want to use. For example, to use the Google Drive V3 client, install the `google-apis-drive_v3` gem.
```
Comment (quartzmo) [inline google-api-client/OVERVIEW.md:6]: What do you think about adding a sentence here mentioning the extracted gems? (`google-apis-core` and `google-apis-generator`?)

File: google-api-client/google-api-client.gemspec (hunk 1 above)
Comment (dazuma) [reply google-api-client/google-api-client.gemspec]: done

File: google-api-client/OVERVIEW.md (hunk 2 above)
Comment (dazuma) [reply google-api-client/OVERVIEW.md]: done

File: google-api-client/OVERVIEW.md (hunk 3 above)
Comment (dazuma) [reply google-api-client/OVERVIEW.md]: The class reference documentation is still there. (Only the handwritten usage documentation was removed.) Updated to clarify.

File: google-api-client/OVERVIEW.md (hunk 4 above)
Comment (dazuma) [reply google-api-client/OVERVIEW.md:6]: Added a sentence regarding generator. Core should just be a dependency (like google-cloud-core), and I don't expect people would install it directly.

**Review by quartzmo — APPROVED** [review]

**Pull request merged**